posthog/hogql_queries/insights/lifecycle_query_runner.py

```diff
@@ -278,13 +279,39 @@ def event_filter(self) -> ast.Expr:
         else:
             return ast.And(exprs=event_filters)
 
+    @property
+    def group_type_index(self) -> int | None:
+        return self.query.aggregation_group_type_index
+
     @cached_property
     def events_query(self):
         with self.timings.measure("events_query"):
+            target_field = "person_id"
+
+            event_filters = [self.event_filter]
+
+            if self.group_type_index is not None:
+                group_index = int(self.group_type_index)
+                if 0 <= group_index <= 4:
+                    target_field = f"$group_{group_index}"
+
+                    event_filters.append(
+                        ast.Not(
+                            expr=ast.Call(
+                                name="has",
+                                args=[
+                                    ast.Array(exprs=[ast.Constant(value="")]),
+                                    ast.Field(chain=["events", f"$group_{self.group_type_index}"]),
+                                ],
+                            ),
+                        ),
+                    )
+
```

Comment:
I think this should go directly into the `self.event_filter()` property. Something else could also access this property on the query runner to get the same event set, and would then get inconsistent results.